src/ng/q.js

```diff
@@ -354,8 +354,8 @@ function qFactory(nextTick, exceptionHandler) {
       var that = this;
       var done = false;
       try {
-        if ((isObject(val) || isFunction(val))) then = val && val.then;
-        if (isFunction(then)) {
+        if(isPromiseLike(val)) {
+          then = val && val.then
```

Comment:
Yeah it's super specific, although that does force implementations to be consistent which might be worth it. The only way to do this would be for `isPromiseLIke` to return the `.then` like you mentioned...